src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8140,130 +8154,198 @@ void checkWellKnown(WellKnownMember member)
-                    originalBinder: this,
-                    diagnose: false,
-                    ref useSiteInfo);
+                Debug.Assert(receiver.Type is not null);
+                var useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics); // TODO2 need to report
+                var lookupResult = LookupResult.GetInstance();
 
-                if (lookupResult.IsMultiViable)
+                if (argIsIndex)
                 {
-                    foreach (var candidate in lookupResult.Symbols)
+                    // Look for `T this[int i]` indexer
+
+                    LookupMembersInType(
+                        lookupResult,
+                        receiver.Type,
+                        WellKnownMemberNames.Indexer,
+                        arity: 0,
+                        basesBeingResolved: null,
+                        LookupOptions.Default,
+                        originalBinder: this,
+                        diagnose: false,
+                        ref useSiteInfo);
+
+                    if (lookupResult.IsMultiViable)
                     {
-                        if (!candidate.IsStatic &&
-                            candidate is PropertySymbol property &&
-                            IsAccessible(property, ref useSiteInfo) &&
-                            property.OriginalDefinition is { ParameterCount: 1 } original &&
-                            original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
+                        foreach (var candidate in lookupResult.Symbols)
                         {
-                            // note: implicit copy check on the indexer accessor happens in CheckPropertyValueKind
-                            patternSymbol = property;
-                            return true;
+                            if (!candidate.IsStatic &&
+                                candidate is PropertySymbol property &&
+                                IsAccessible(property, ref useSiteInfo) &&
+                                property.OriginalDefinition is { ParameterCount: 1 } original &&
+                                original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
+                            {
+                                diagnostics.Add(syntax, useSiteInfo);
+
+                                var intPlaceholder = new BoundIndexOrRangeIndexerPatternValuePlaceholder(syntax, Compilation.GetSpecialType(SpecialType.System_Int32)) { WasCompilerGenerated = true };
+                                argumentPlaceholders = ImmutableArray.Create(intPlaceholder);
+
+                                var analyzedArguments = AnalyzedArguments.GetInstance();
+                                analyzedArguments.Arguments.Add(intPlaceholder);
+                                var properties = ArrayBuilder<PropertySymbol>.GetInstance();
+                                properties.AddRange(property);
```

Comment:
>property [](http://example.com/codeflow?start=52&length=8)

Is ```BindIndexerOrIndexedPropertyAccess``` going to report use-site info for this symbol?

Reply:
Yes, as part of overload resolution: `ReportUseSiteInfo(results, ref useSiteInfo);` (in `PerformMemberOverloadResolution`)